### PR TITLE
Fixing  `same-project` onboarding

### PR DESCRIPTION
**Pull request by Noamstrauss**
- Adjusting `same-project` onboarding to use project iam roles instead of organization iam roles
- Adding `pr-checks.yaml` workflow
- Adding `CODEOWNERS` 
- Updating `README.md`
- Updating `examples`

### CODE REVIEW

File: .github/workflows/pr-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: PR Checks
```
[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:1]: Added `gitleaks` to `trivy-scan.yaml`

[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:1]: Added `gitleaks` to `trivy-scan.yaml`

File: .github/workflows/pr-checks.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  pr-checks:
+    name: Terraform Validation
```
[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:8]: Skipping

File: .github/workflows/pr-checks.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:1]: Skipping

[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:1]: Skipping

[the remark this answers is not in the document]
Comment (Noamstrauss) [reply .github/workflows/pr-checks.yaml:1]: Skipping

**Review by iliasokolovski192 — APPROVED** [review]

**Pull request closed without merging**